components/core/DataView.js

```diff
@@ -248,20 +221,44 @@ export default class DataView extends React.Component {
     window.removeEventListener("remote-data-deletion", this._handleDataDeletion);
     window.removeEventListener("remote-slate-object-remove", this._handleRemoteSlateObjectRemove);
     window.removeEventListener("remote-slate-object-add", this._handleRemoteSlateObjectAdd);
+    window.removeEventListener("scroll", this._handleCheckScroll);
     window.removeEventListener("remote-update-carousel", this._handleUpdate);
   }
 
-  _increment = (direction) => {
-    if (
-      direction > 0 &&
-      this.state.startIndex + VIEW_LIMIT < this.props.viewer.library[0].children.length
-    ) {
-      this.setState({ startIndex: this.state.startIndex + VIEW_LIMIT });
-    } else if (direction < 0 && this.state.startIndex - VIEW_LIMIT >= 0) {
-      this.setState({ startIndex: this.state.startIndex - VIEW_LIMIT });
+  _handleDebounce = (func, wait) => {
+    let timeout;
+
+    return function executedFunction(...args) {
+      const later = () => {
+        clearTimeout(timeout);
```

Comment:
Not super critical, but you don't really need this `clearTimeout` here b/c the only time `later` will be called is at the end of the time out. clearTimeout is only needed if you want to cancel the timer before the time is up